torchvision/datasets/cityscapes.py

```diff
@@ -109,22 +109,19 @@ def __init__(self, root, split='train', mode='fine', target_type='instance',
         self.images = []
         self.targets = []
 
-        if mode not in ['fine', 'coarse']:
-            raise ValueError('Invalid mode! Please use mode="fine" or mode="coarse"')
-
-        if mode == 'fine' and split not in ['train', 'test', 'val']:
-            raise ValueError('Invalid split for mode "fine"! Please use split="train", split="test"'
-                             ' or split="val"')
-        elif mode == 'coarse' and split not in ['train', 'train_extra', 'val']:
-            raise ValueError('Invalid split for mode "coarse"! Please use split="train", split="train_extra"'
-                             ' or split="val"')
+        verify_str_arg(mode, ("fine", "coarse"), "mode")
+        if mode == "fine":
+            # TODO: add condition "invalid argument of mode == fine ..."
```

Comment:
maybe make `msg` an optional argument in `verify_str_arg`, which defaults to the current message if not passed?